pacsman/filesystem_dev_client.py

```diff
@@ -175,19 +176,29 @@ def fetch_thumbnail(self, study_id: str, series_id: str) -> Optional[str]:
 
         series_items = sorted(series_items, key=lambda t: t[1].SOPInstanceUID)
 
-        thumbnail_series_path = series_items[len(series_items) // 2][0]
-        shutil.copy(thumbnail_series_path, self.dicom_dir)
-
-        thumbnail_filename = os.path.basename(thumbnail_series_path)
-        dcm_path = os.path.join(self.dicom_dir, thumbnail_filename)
-        try:
-            thumbnail_ds = dcmread(dcm_path)
-            png_path = os.path.splitext(dcm_path)[0] + '.png'
-            process_and_write_png(thumbnail_ds, png_path)
-        finally:
-            os.remove(dcm_path)
+        thumbnail_index = len(series_items) // 2
+        thumbnail_archive_path = series_items[thumbnail_index][0]
+        thumbnail_instance_id = series_items[thumbnail_index][1].SOPInstanceUID
+
+        # copying to instance ID ensures that the filename is unique
+        dcm_path = os.path.join(self.dicom_dir, f'{thumbnail_instance_id}.dcm')
+        shutil.copy(thumbnail_archive_path, dcm_path)
+
+        png_path = process_and_write_png_from_file(dcm_path)
         return png_path
 
+    def fetch_slice_thumbnail(self, study_id: str, series_id: str,
```

Comment:
Is it too much of a performance penalty to use this method in `fetch_thumbnail` because you have to iterate over the datasets twice-- once to get the middle slice instance ID and another in this method?

If the performance penalty doesn't make sense then that's fine-- just thinking that you could make the `ds -> png_path` conversion a bit more DRY that way. Or perhaps just encapsulating that part of the common functionality between the two methods would work?

Reply:
It still only loops through the big/main `dicom_datasets` once in `fetch_thumbnail`, it just goes through the series instances after for the middle.